openfl/experimental/runtime/local_runtime.py

```diff
@@ -1,232 +1,613 @@
+        def __init__(self, collaborator_actor, collaborator, f_name) -> None:
+            self.f_name = f_name
+            self.collaborator_actor = collaborator_actor
+            self.collaborator = collaborator
+            self.f = lambda *args, **kwargs,: self.collaborator_actor.execute_from_col.remote(
+                self.collaborator, self.f_name, *args, **kwargs
+            )
+
+        def remote(self, *args, **kwargs):
+            return self.f(*args, *kwargs)
+                
+    collaborator_ray_refs = []
+
+    times_called = 0
+    # logic to sort collaborators by gpus, if collaborators have the same number of gpu then they are sorted by cpu
+    cpu_magnitude = len(str(abs(max([i.num_cpus for i in collaborators]))))
+    min_gpu = min([i.num_gpus for i in collaborators])
+    min_gpu = max(min_gpu,0.0001)
+    collaborators_sorted_by_gpucpu = sorted(collaborators, key=lambda x:x.num_gpus/min_gpu*10**cpu_magnitude+x.num_cpus)
+    initializations = []
+    
+    for collaborator in collaborators_sorted_by_gpucpu:
+        #initialize actor group
+        if times_called % collaborators_per_group == 0:
+            max_num_cpus = max([i.num_cpus for i in collaborators_sorted_by_gpucpu[times_called:times_called+collaborators_per_group]])
+            max_num_gpus = max([i.num_gpus for i in collaborators_sorted_by_gpucpu[times_called:times_called+collaborators_per_group]])
+            print(f'creating actor with {max_num_cpus}, {max_num_gpus}')
+            collaborator_actor = (
+                ray.remote(RayGroup)
+                .options(num_cpus=max_num_cpus, num_gpus=max_num_gpus, max_concurrency=max_concurrency)
+                .remote()
+            )
+        # add collaborator to actor group
+        initializations.append(collaborator_actor.append.remote(
+            collaborator.get_name(),
+            private_attributes_callable=collaborator.private_attributes_callable,
+            **collaborator.kwargs,
+        ))
+
+        times_called += 1
+
+        # append GroupMember to output list
+        collaborator_ray_refs.append(
+            GroupMember(collaborator_actor, collaborator.get_name())
+        )
+    # Wait for all collaborators to be created on actors 
+    ray.get(initializations)
+    
+
+    return collaborator_ray_refs
+
+
+class RayGroup:
+    # Ray actor that contains a group of collaborators, has funtion execute_from_col that takes a name of collaborator, funtion name and *args, **kwargs, to run the given funtion in the given collaborator
+    
+    def __init__(self):
+        self.collaborators = {}
+
+    def append(
+        self,
+        name: str = "",
+        private_attributes_callable: Callable = None,
+        **kwargs,
+    ):
+        interface_module = importlib.import_module("openfl.experimental.interface")

```

Comment:
This can again be simplified to:

`from openfl.experimental.interface import Collaborator`

Reply:
Changed to from openfl.experimental.interface import Collaborator